main.go

```diff
@@ -82,9 +86,16 @@ func parseJSON(filename string) error {
 *	The main difference between types of loggers is the label (eg INFO) and additional data provided .
  */
 func initLogging() (*os.File, error) {
+	handlerEnv, err := settings.GetHandlerEnvironment()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to open handler environment")
+	}
+
+	logfile = handlerEnv.HandlerEnvironment.LogFolder + "/" + logfileLogName
+
 	file, err := os.OpenFile(logfile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Failed to create/open specified log file")
```

Comment:
NIT: Add the filename of the log in the error message
Essentially : `"Failed to create/open guest-agent-test-extension.log file"`